### ISSUE TITLE
Add method to programatically set options

### ISSUE THREAD

**Issue by derweili** [pluginkollektiv/cachify#199]
The plugin options are stored in the Database and the default values are "hardcoded" into the `_get_options()`method:
https://github.com/pluginkollektiv/cachify/blob/develop/inc/cachify.class.php

https://github.com/pluginkollektiv/cachify/blob/dcac6b7babdef4454782faf6f9c076699cb11944/inc/cachify.class.php#L455-L470

There is currently no filter to overwrite the options or default options.

Depending on the WordPress setup there could be a need to set these options programmatically, e.g. within a large Multisite Installation where you don't want to set the options for each site manually.

Basically, there are two ways of how such a "programmatically" functionality could be setup.

1. We could add a filter before we return from the `_get_options()`.
2. Using PHP Constants: Some plugin provide PHP constants that can be used to overwrite the options. Those constants are normally set inside the `wp-config.php` file. Example from the WP Mail SMTP Plugin: https://wpmailsmtp.com/docs/how-to-secure-smtp-settings-by-using-constants/

I'm not sure which solution is better, we could also implement both functionalities.

**Comment by bueltge** [comment]
I mean we should add a filter hook, because it is a plugin and should change, enhance this also with a small plugin. Constants are helpful, but in this case a overhead for the storage and not clean in the namespace about the whole installation of the client side.

**Comment by chesio** [comment]
Note that one can always filter the return value of [get_option](https://developer.wordpress.org/reference/functions/get_option/) like this:

```php
add_filter('option_cachify', function (array $settings): array {
    $settings['cache_expires'] = 24;
    return $settings;
}, 10, 1);
```

**Comment by bueltge** [comment]
Right, you have right, great hint. Each option in the WP space has the option to filter with the hook [`option_`](https://developer.wordpress.org/reference/hooks/option_option/) or [`pre_option_`](https://developer.wordpress.org/reference/hooks/pre_option_option/). So maybe we should only add this to the documentation, without changes at the plugin in this context. @derweili helpful for your problem?

**Comment by derweili** [comment]
@chesio totally missed that. Maybe this should be added to the documentation.
